mesonbuild/compilers/rust.py

```diff
@@ -196,6 +196,20 @@ def get_debug_args(self, is_debug: bool) -> T.List[str]:
     def get_optimization_args(self, optimization_level: str) -> T.List[str]:
         return rust_optimization_args[optimization_level]
 
+    def build_rpath_args(self, env: 'Environment', build_dir: str, from_dir: str,
+                         rpath_paths: T.Tuple[str, ...], build_rpath: str,
+                         install_rpath: str) -> T.Tuple[T.List[str], T.Set[bytes]]:
+        args, to_remove = super().build_rpath_args(env, build_dir, from_dir, rpath_paths,
+                                                   build_rpath, install_rpath)
+
+        # ... but then add rustc's sysroot to account for rustup
+        # installations
+        rustc_rpath_args = []
+        for arg in args:
+            rustc_rpath_args.append('-C')
+            rustc_rpath_args.append('link-arg=' + arg + ':' + self.get_target_libdir())
```

Comment:
Maybe other people have different opinions here, but I've always found this form harder to read than either a f-string or a str.format(). I think this case propbably tips past Eli's preference for a `str.format`, so I'd appreciate that if possible.

Reply:
I agree for what it's worth; in this case the `... + ...` form is preexisting, only moved from elsewhere in the code. I can change it to an f-string, perhaps in a separate commit?

Reply:
Yeah, a separate commit would be fine.

Reply:
```python
f'link-arg={arg}:{self.get_target_libdir()}'
```

is reasonably comprehensible, yeah. Simple statements, no nested quotes. Although the code segment is the majority of the string, it's not overwhelmingly so (like some cases where it's entirely grammar + code).